elide-core/src/main/java/com/yahoo/elide/parsers/state/RelationshipTerminalState.java

```diff
@@ -84,7 +84,38 @@ public RelationshipTerminalState(PersistentResource record, String relationshipN
 
     @Override
     public Supplier<Pair<Integer, JsonNode>> handlePatch(StateContext state) {
-        return handleRequest(state, this::patch);
+        return handlePatchRequest(state, this::patch);
+    }
+
+    private Supplier<Pair<Integer, JsonNode>> handlePatchRequest(StateContext state,
```

Comment:
Rather than duplicating the logic, it may make most sense here to modify the existing `handleRequest` method. Namely, either add a parameter that determines the type of response or just check if the `handler == this::patch` for a short-term solution since that seems to be the only use case right now.